cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/StreamToDataset.java

```diff
@@ -79,8 +80,9 @@ public void initialize() throws Exception {
     }
     LOG.info("Using '{}' as the input stream and '{}' as the output dataset.", inputStream, outputDataset);
     job.setMapperClass(mapper);
-    context.addInput(Input.ofStream(inputStream));
-    context.addOutput(Output.ofDataset(outputDataset));
+    dataNamespace = dataNamespace == null ? getContext().getNamespace() : dataNamespace;
+    context.addInput(Input.ofStream(inputStream).fromNamespace(dataNamespace));
+    context.addOutput(Output.ofDataset(outputDataset).fromNamespace(dataNamespace));
```

Comment:
OK...but why not put lines 64 and 83 together, as they are the same two, and are repeated a total of four times? I'm not saying you need (necessarily) to factor them to another method, but perhaps the definition of dataNamespace should always be together in one place:
```
String dataNamespace = context.getRuntimeArguments().get(WikipediaPipelineApp.NAMESPACE_ARG);
dataNamespace = dataNamespace == null ? getContext().getNamespace() : dataNamespace;